### PR TITLE
*: fix the issue that loadcluster does not remove overlap regions

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
fix https://github.com/pingcap/pd/issues/2011

### What is changed and how it works?
check the region epoch firstly.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/core/region.go [hunk 1]
```diff
@@ -304,6 +304,9 @@ func (r *RegionInfo) GetID() uint64 {
 
 // GetMeta returns the meta information of the region.
 func (r *RegionInfo) GetMeta() *metapb.Region {
+	if r == nil {
+		return nil
+	}
```
Comment (disksing) [inline server/core/region.go]: why need this?

Comment (nolouch) [reply server/core/region.go]: If not will panic in this [line](https://github.com/pingcap/pd/pull/2022/files#diff-02ca8f1595ce93b75591586acd227047R244).

File: server/cluster/cluster.go [hunk 2]
```diff
@@ -242,7 +237,17 @@ func (c *RaftCluster) LoadClusterInfo() (*RaftCluster, error) {
 
 	start = time.Now()
 
-	if err := c.storage.LoadRegions(c.core.PutRegion); err != nil {
+	// used to load region from kv storage to cache storage.
+	putRegionToCache := func(region *core.RegionInfo) []*core.RegionInfo {
```
Comment (lhy1024) [inline server/cluster/cluster.go]: how about `putRegion`?

File: tests/server/region_syncer/region_syncer_test.go [hunk 3]
```diff
@@ -94,6 +94,13 @@ func (s *serverTestSuite) TestRegionSyncer(c *C) {
 		err = rc.HandleRegionHeartbeat(region)
 		c.Assert(err, IsNil)
 	}
+	// merge case
```
Comment (lhy1024) [inline tests/server/region_syncer/region_syncer_test.go]: do we need to add test with different merge order and epoch?

File: tests/server/region_syncer/region_syncer_test.go [hunk 4]
```diff
@@ -94,6 +94,26 @@ func (s *serverTestSuite) TestRegionSyncer(c *C) {
...
+	// region2 -> region1 -> region0
+	regions[0] = regions[0].Clone(core.WithEndKey(regions[2].GetEndKey()), core.WithIncVersion(), core.WithIncVersion())
+	err = rc.HandleRegionHeartbeat(regions[2])
+	c.Assert(err, IsNil)
+
+	// merge case
```
Comment (lhy1024) [inline tests/server/region_syncer/region_syncer_test.go]: do we need to add some comments about these cases?

Comment (nolouch) [reply tests/server/region_syncer/region_syncer_test.go]: I have comment `region3 -> region4`, the direct is `->`. what your concerns?

Comment (lhy1024) [reply tests/server/region_syncer/region_syncer_test.go]: with `epoch`?

Comment (nolouch) [reply tests/server/region_syncer/region_syncer_test.go]: PTAL

**Pull request merged**